### PR TITLE
LPS-131184 Update editing languageId in the data engine side when change languageId in TranslationsManager

### PR DESCRIPTION
**Pull request by interaminense**
- [LPS-131185](https://issues.liferay.com/browse/LPS-131185) A message regarding detectClickOutside is being displayed on devTools console
- [LPS-131184](https://issues.liferay.com/browse/LPS-131184) The changes on languages are not being reflected on Form Builder
- [LPS-131200](https://issues.liferay.com/browse/LPS-131200) User cannot save a form on Web Content

cc: @miltonmc, @alinedoleron

### CODE REVIEW

**Comment by interaminense** [comment]
ci:test:sf

**Comment by interaminense** [comment]
ci:test:relevant

**Comment by interaminense** [comment]
ci:reopen

**Comment by interaminense** [comment]
ci:test:relevant

**Comment by interaminense** [comment]
ci:reopen

**Comment by interaminense** [comment]
ci:test:relevant

**Comment by interaminense** [comment]
ci:reopen

**Comment by interaminense** [comment]
ci:test:relevant

**Review by miltonmc — CHANGES_REQUESTED** [review]

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/data-engine/DataEngineLayoutBuilderHandler.es.js [hunk 1]
```diff
@@ -12,23 +12,13 @@
...
- * Software Foundation; either version 2.1 of the License, or (at your option)
- * any later version.
- *
- * This library is distributed in the hope that it will be useful, but WITHOUT
- * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
- * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
- * details.
- */
+const clickOutside = (target, ...elements) =>
+	!elements.some((element) => !!target.closest(element));
```
[the remark this answers is not in the document]
Comment (interaminense) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/data-engine/DataEngineLayoutBuilderHandler.es.js:16]: That's true. I updated it. Thanks!

File: modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/data-engine/DataEngineLayoutBuilderHandler.es.js [hunk 2]
```diff
@@ -54,83 +44,88 @@ export default function ({defaultLanguageId, namespace}) {
...
 		if (
-			!event.target.closest(
-				'.ddm-form-builder-wrapper, .multi-panel-sidebar, .dropdown-menu'
-			)
+			clickOutside(target, [
+				'.ddm-form-builder-wrapper',
+				'.multi-panel-sidebar',
+				'.lfr-icon-menu-open',
+				'.input-localized-content',
+			])
```
[the remark this answers is not in the document]
Comment (interaminense) [reply modules/apps/document-library/document-library-web/src/main/resources/META-INF/resources/document_library/js/data-engine/DataEngineLayoutBuilderHandler.es.js:100]: Good catch! I update it. Thanks

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/DataEngineLayoutBuilderHandler.es.js [hunk 3]
```diff
@@ -12,9 +12,13 @@
  * details.
  */
 
-import {DataConverter} from 'data-engine-taglib';
+const clickOutside = (target, ...elements) =>
+	!elements.some((element) => !!target.closest(element));
```
[the remark this answers is not in the document]
Comment (interaminense) [reply modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/DataEngineLayoutBuilderHandler.es.js:16]: Done

File: modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/DataEngineLayoutBuilderHandler.es.js [hunk 4]
```diff
@@ -69,89 +73,93 @@ export default function ({defaultLanguageId, namespace}) {
...
+	const detectClickOutside = async ({target}) => {
 		if (
-			!event.target.closest('.ddm-form-builder-wrapper') &&
-			!event.target.closest('.multi-panel-sidebar')
+			clickOutside(target, [
+				'.ddm-form-builder-wrapper',
+				'.multi-panel-sidebar',
+				'.lfr-icon-menu-open',
+				'.input-localized-content',
+			])
```
[the remark this answers is not in the document]
Comment (interaminense) [reply modules/apps/journal/journal-web/src/main/resources/META-INF/resources/js/DataEngineLayoutBuilderHandler.es.js:131]: Done

**Comment by interaminense** [comment]
ci:test:sf

**Comment by interaminense** [comment]
ci:test:sf

**Comment by interaminense** [comment]
ci:test:relevant

**Comment by interaminense** [comment]
ci:test:relevant

**Comment by interaminense** [comment]
@miltonmc, this PR is able to merge with `feature flag branch` https://github.com/liferay-forms/liferay-portal/pull/197

**Review by miltonmc — APPROVED** [review]

**Pull request merged**